cmd/soroban-cli/src/commands/contract/deploy/wasm.rs

```diff
@@ -100,19 +108,83 @@ pub enum Error {
     Network(#[from] network::Error),
     #[error(transparent)]
     Wasm(#[from] wasm::Error),
+    #[error("cannot access config dir for alias file")]
+    CannotAccessConfigDir,
+    #[error(
+        "alias must be 1-30 chars long, and have only letters, numbers, underscores and dashes"
+    )]
+    InvalidAliasFormat { alias: String },
+    #[error(transparent)]
+    JsonSerialization(#[from] serde_json::Error),
+    #[error(transparent)]
+    Io(#[from] std::io::Error),
 }
 
 impl Cmd {
     pub async fn run(&self) -> Result<(), Error> {
+        self.validate_alias()?;
+
         let res = self.run_against_rpc_server(None, None).await?.to_envelope();
         match res {
             TxnEnvelopeResult::TxnEnvelope(tx) => println!("{}", tx.to_xdr_base64(Limits::none())?),
             TxnEnvelopeResult::Res(contract) => {
+                self.save_contract_id(&contract)?;
                 println!("{contract}");
             }
         }
         Ok(())
     }
+
+    fn validate_alias(&self) -> Result<(), Error> {
+        match self.alias.clone() {
+            Some(alias) => {
+                let regex = Regex::new(r"^[a-zA-Z0-9_-]{1,30}$").unwrap();
+
+                if regex.is_match(&alias) {
+                    Ok(())
+                } else {
+                    Err(Error::InvalidAliasFormat { alias })
+                }
+            }
+            None => Ok(()),
+        }
+    }
+
+    fn alias_path_for(&self, alias: &str) -> Result<PathBuf, Error> {
+        let config_dir = self.config.config_dir()?;
+        let network = self.config.network.network.clone().expect("must be set");
```

Comment:
This is the case of the RPC failures since a lot of tests don't use a network alias, rather the rpc_url and network_passphrase directly. I think the better solution is to use `config.get_network()`, which returns the resolved network (the name is used to look it up). Then you could use the network passphrase to determine which network it is. Perhaps you could even add it to the `AliasData` type. e.g.

```rust
struct AliasData {
    ids: BTreeMap<String, String>;
}
```
This would then map the network passphrase to ids, then you could have the contract-ids folder be flat.

Reply:
Good call. Leigh made a comment somewhere saying that arbitrary networks wouldn't work as well, so the fair assumption we can make (and state in the docs), is that alias work for `testnet`, `pubnet`, `futurenet` and `local`.

Reply:
We could do `alias.testnet` etc. However, the network passphrases are consistent so would be an okay key to use in the map IMO. @leighmcculloch might think otherwise though (I normally avoid spaces in string keys), but we still could also map the network passphrase to `testnet` and use those as keys.

Reply:
Arbitrary networks would work fine for this. I see right now the alias is stored in file inside a 'network' directory. I think you can store aliases in a single namespace for all networks. This is similar to how we store aliases for keys across all networks. Then store the network _passphrase_ not the network name, in the .json for the alias. If anywhere we use the alias we use it with a network with a passphrase not matching the recorded network passphrase, we can show an error.

Reply:
I would avoid a map where a single aliases can be mapped to contracts on different networks. It feels like it steps over a complexity boundary where it starts to become harder for a user to grasp what's happening and how it works. A simple 1-1 where alias maps to contract ID on one network is easier to understand. I understand it is not as powerful, but we can make this more powerful in the future if evidence shows it's a worthy tradeoff.